### PR TITLE
fix(op_crates/fetch): ensure Request.method to use the default value if not defined

### PR DESCRIPTION
**Pull request by kidonng**
Ensure `Request.method` to be the default value (`GET`) if `init.method` is not defined, which follows browser's behavior.

Originally sindresorhus/ky#292

```ts
// Before this PR
console.log(new Request('', { method:undefined }).method) // undefined

// After this PR
console.log(new Request('', { method:undefined }).method) // 'GET'
```

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]
LGTM, great fix @kidonng

File: op_crates/fetch/26_fetch.js [hunk 1]
```diff
@@ -990,7 +990,7 @@
         this.url = new URL(String(input)).href;
       }
 
-      if (init && "method" in init) {
+      if (init && "method" in init && init.method) {
```
Comment (mayankchoubey1) [inline op_crates/fetch/26_fetch.js:993]: Shouldn't it be if (init && init.method) ?

**Review by mayankchoubey1 — COMMENTED** [review]
There are extra checks in the if condition.

**Review by mayankchoubey1 — COMMENTED** [review]
The if condition has an extra check

File: op_crates/fetch/26_fetch.js (hunk 1 above)
Comment (kidonng) [reply op_crates/fetch/26_fetch.js:993]: Ideally we can just check `init.method`, I'm just following existing code like https://github.com/denoland/deno/blob/3fc4df3887d1611aeb87b8d1fe08b227b7e724a9/op_crates/fetch/26_fetch.js#L998-L1000

**Pull request merged**